lib/dialects/abstract/connection-manager.js

```diff
@@ -184,11 +184,28 @@ ConnectionManager.prototype.getConnection = function(options) {
   var self = this;
   options = options || {};
 
-  return new Promise(function (resolve, reject) {
-    self.pool.acquire(function(err, connection) {
-      if (err) return reject(err);
-      resolve(connection);
-    }, options.priority, options.type, options.useMaster);
+  return Promise.resolve().then(function () {
+    if (self.sequelize.options.databaseVersion === 0) {
+      return self.$connect(self.config).then(function (connection) {
+        var _options = {};
+        _options.transaction = { connection: connection }; // Cheat .query to use our private connection
+        _options.logging = function () {};
+        _options.logging.__testLoggingFn = true;
+
+        return self.sequelize.databaseVersion(_options).then(function (version) {
+          self.sequelize.options.databaseVersion = version;
+
+          self.$disconnect(connection);
+        });
+      });
+    }
+  }).then(function () {
```

Comment:
`return;` would not be returning something, i don't really see the importance in any case, but up to @janmeier